tests/baselines/reference/thisTypeInTuples.types

```diff
@@ -16,24 +16,24 @@ let t: [number, string] = [42, "hello"];
 let a = t.slice();
 >a : [number, string]
 >t.slice() : [number, string]
->t.slice : { (start?: number, end?: number): (string | number)[]; (): [number, string]; }
+>t.slice : { (start?: number, end?: number): [number, string]; (): [number, string]; }
 >t : [number, string]

```

Comment:
Wrong, concat and slice methods must not return `this` type when it is a tuple type. It cannot avoid.